src/query/Index.cpp

```diff
@@ -271,8 +271,17 @@ ProjectionMatch Index::findCellOrProjection(const Eigen::VectorXd &location, int
     // If no triangle is found, fall-back on NP
     return findNearestProjection(location, n);
   } else {
-    PRECICE_UNREACHABLE("Volume coupling 3D not  implemented");
-    return findTriangleProjection(location, n);
+
+    // Find correct tetra, or fall back to NP
+    auto matchedTetra = getEnclosingTetrahedra(location);
+    for (const auto &match : matchedTetra) {
+      // Matches are raw indices, not (indices, distance) pairs
+      auto polation = mapping::Polation(location, _mesh->tetrahedra()[match]);
+      if (polation.isInterpolation()) {
+        return {polation, 0.0};
```

Comment:
Could you comment why you are returning zero here instead of the distance? You wrote it's only the raw index without the distance. Is the zero supposed to be a dummy value? Could we get rid of it?

Reply:
The comment needs improvement. The location is within the cell, hence a distance of 0.

Reply:
The comment above just explains why we use mesh->tetrahedra()[match] instead of the match.first like with triangles. (This comes from the nature of the query)
I'll add another comment
Zero is indeed because we are inside, and non-zero values occur only with fallback to NP